OsmAnd-java/src/main/java/net/osmand/search/core/SearchCoreFactory.java

```diff
@@ -735,6 +737,44 @@ protected static class PoiTypeResult {
 		public Set<String> foundWords = new LinkedHashSet<String>();
 	}
 
+	public static  class SearchAmenityBrandAPI extends SearchBaseAPI {
+		@Override
```

Comment:
1. This is perfectly part of SearchAmenityTypesAPI  - addPoiTypeResult. 
2. I'm not sure search code should depend on brands, it should be generic poisubtypes (https://github.com/osmandapp/OsmAnd/issues/16397) in task it doesn't specify only brand is problem. Could be operator or any searchable field.